src/components/Login.jsx

```diff
@@ -42,7 +46,7 @@ const Login = () => {
       setAuth({ user, pwd, accessToken });
```

Comment:
roles prop is not provided to setAuth(), that's why auth?.roles is undefined in RequiredAuth.jsx component